internal/clientconn/listener.go

```diff
@@ -105,17 +106,30 @@ func (l *Listener) Run(ctx context.Context) error {
 
 		// run connection
 		go func() {
+			connID := fmt.Sprintf("%s -> %s", netConn.RemoteAddr(), netConn.LocalAddr())
+
+			runCtx, runCancel := ctxutil.WithDelay(ctx.Done(), delay)
+			defer runCancel()
+
+			if l.opts.TestConnTimeout != 0 {
+				runCtx, runCancel = context.WithTimeout(runCtx, l.opts.TestConnTimeout)
+				defer runCancel()
+			}
+
+			defer pprof.SetGoroutineLabels(runCtx)
```

Comment:
Ok, I guess you tested it :) 
(I mean, the fact that we both call `SetGoroutineLabels` right here and also defer it)

Reply:
It works because `runCtx` is evaluated before `pprof.SetGoroutineLabels` call is put on stack